Mathlib/Algebra/Module/Submodule/Localization.lean

```diff
@@ -113,3 +118,50 @@ instance IsLocalizedModule.toLocalizedQuotient' (M' : Submodule R M) :
 
 instance (M' : Submodule R M) : IsLocalizedModule p (M'.toLocalizedQuotient p) :=
   IsLocalizedModule.toLocalizedQuotient' _ _ _ _
+
+section LinearMap
+
+variable {P : Type w} [AddCommGroup P] [Module R P]
+variable {Q : Type w'} [AddCommGroup Q] [Module R Q] [Module S Q] [IsScalarTower R S Q]
+variable (f' : P →ₗ[R] Q) [IsLocalizedModule p f']
+
+lemma LinearMap.localized'_ker_eq_ker_localizedMap (g : M →ₗ[R] P) :
+    Submodule.localized' S p f (LinearMap.ker g) =
+      LinearMap.ker ((IsLocalizedModule.map p f f' g).extendScalarsOfIsLocalization p S) := by
+  ext x
+  simp only [Submodule.mem_localized', mem_ker, extendScalarsOfIsLocalization_apply']
+  constructor
+  · rintro ⟨m, hm, a, ha, rfl⟩
+    rw [IsLocalizedModule.map_mk', hm]
+    simp
+  · intro h
+    obtain ⟨⟨a, b⟩, rfl⟩ := IsLocalizedModule.mk'_surjective p f x
+    simp only [Function.uncurry_apply_pair, IsLocalizedModule.map_mk',
+      IsLocalizedModule.mk'_eq_zero, IsLocalizedModule.eq_zero_iff p f'] at h
+    obtain ⟨c, hc⟩ := h
+    refine ⟨c • a, by simpa, c * b, by simp⟩
+
+/--
+The canonical map from the kernel of `g` to the kernel of `g` localized at a submonoid.
+
+This is a localization map by `LinearMap.toKerLocalized_isLocalizedModule`.
+-/
+@[simps]
+def LinearMap.toKerIsLocalized (g : M →ₗ[R] P) :
+    ker g →ₗ[R] ker (IsLocalizedModule.map p f f' g) where
+  toFun x := ⟨f x, by simp [LinearMap.mem_ker, LinearMap.mem_ker.mp x.property]⟩
+  map_add' x y := by
+    simp only [AddSubmonoid.coe_add, Submodule.coe_toAddSubmonoid, map_add, AddSubmonoid.mk_add_mk]
+  map_smul' a x := by
+    simp only [SetLike.val_smul, LinearMapClass.map_smul, RingHom.id_apply, SetLike.mk_smul_mk]
```

Comment:
```suggestion
    ker g →ₗ[R] ker (IsLocalizedModule.map p f f' g) :=
  f.restrict (fun x hx ↦ by simp [LinearMap.mem_ker, LinearMap.mem_ker.mp hx])
```
Not tested

Reply:
Works as is! (Irrelevant side node: Lean needs a `noncomputable` with your version (since `IsLocalizedModule.map` is `noncomputable`), but not with the explicit one. Is this expected?)